pkg/testutils/kubernetes/builders/pod.go

```diff
@@ -57,6 +57,12 @@ func (b *podBuilder) WithContainer(c corev1.Container) PodBuilder {
 }
 
 func (b *podBuilder) Build() *corev1.Pod {
+	containers := b.containers
+
+	if containers == nil {
+		containers = defaultContainers()
+	}
```

Comment:
This function is used in many tests. I don't think ti should be too opinionated as to defined some default containers.
If you want to simply the setup of some specific tests (for instance, the pod helper tests), create a build container function there.